stonesoup/plotter.py

```diff
@@ -1302,11 +1339,51 @@ def plot_tracks(self, tracks, mapping, uncertainty=False, particle=False, track_
                     text=[self._format_state_text(state) for state in track],
                     **scatter_kwargs)
 
+            elif self.dimension == 3:  # plot 3D tracks
+
+                if particle:
+                    raise NotImplementedError
+
+                # create empty error arrays
+                err_x = np.array([np.nan for _ in range(len(track))], dtype=float)
+                err_y = np.array([np.nan for _ in range(len(track))], dtype=float)
+                err_z = np.array([np.nan for _ in range(len(track))], dtype=float)
+
+                if uncertainty:  # find x,y,z error bars for relevant states
+
+                    for count, state in enumerate(track):
+
+                        if not count % err_freq:  # ie count % err_freq = 0
+                            HH = np.eye(track.ndim)[mapping, :]  # Get position mapping matrix
+                            cov = HH @ state.covar @ HH.T
+
+                            err_x[count] = np.sqrt(cov[0, 0])
+                            err_y[count] = np.sqrt(cov[1, 1])
+                            err_z[count] = np.sqrt(cov[2, 2])
+
+                self.fig.add_scatter3d(
+                    x=[float(getattr(state, 'mean', state.state_vector)[mapping[0]])
+                       for state in track],
+                    error_x=dict(type='data', thickness=10, width=3, array=err_x),
+
+                    y=[float(getattr(state, 'mean', state.state_vector)[mapping[1]])
+                       for state in track],
+                    error_y=dict(type='data', thickness=10, width=3, array=err_y),
+
+                    z=[float(getattr(state, 'mean', state.state_vector)[mapping[2]])
+                       for state in track],
+                    error_z=dict(type='data', thickness=10, width=3, array=err_z),
+                    # note that 3D error thickness seems to be broken in Plotly
+
+                    text=[self._format_state_text(state) for state in track],
+                    **scatter_kwargs)
+
             track_colors[track] = (self.fig.data[-1].line.color
                                    or self.fig.data[-1].marker.color
                                    or self.get_next_color())
 
-        if uncertainty:
+        # earlier checking means this only applies to 2D.
+        if uncertainty and self.dimension == 2:
```

Comment:
I was hesitant to do that because the ``dimension==2`` line sits within a for-loop that goes through each track whereas uncertainty plotting is outside that. The current implementation plots all the tracks at once and then all the uncertainty at once. My worry is that changing the logic order to track, uncertainty for track, track2, uncertainty for track2, etc. may cause some unintended consequences. It's probably fine but I haven't tried it.